app/routes/datasets/queries.py

```diff
@@ -315,6 +328,85 @@ async def query_dataset_csv_post(
     return CSVStreamingResponse(iter([csv_data.getvalue()]), download=False)
 
 
+@router.post(
+    "/{dataset}/{version}/query/batch",
+    response_class=ORJSONResponse,
+    response_model=UserJobResponse,
+    tags=["Query"],
+    status_code=202,
+)
+async def query_dataset_list_post(
+    *,
+    dataset_version: Tuple[str, str] = Depends(dataset_version_dependency),
+    request: QueryBatchRequestIn,
+    api_key: APIKey = Depends(get_api_key),
+):
+    """Execute a READ-ONLY SQL query on the given dataset version (if
+    implemented)."""
+
+    dataset, version = dataset_version
+
+    default_asset: AssetORM = await assets.get_default_asset(dataset, version)
+    if default_asset.asset_type != AssetType.raster_tile_set:
+        raise HTTPException(
+            status_code=400,
+            detail="Querying on lists is only available for raster tile sets.",
+        )
+
+    if request.feature_collection:
+        for feature in request.feature_collection.features:
+            if (
+                feature.geometry.type != "Polygon"
+                and feature.geometry.type != "MultiPolygon"
+            ):
+                raise HTTPException(
+                    status_code=400,
+                    detail="Feature collection must only contain Polygons or MultiPolygons for raster analysis",
+                )
+
+    job_id = uuid.uuid4()
+
+    # get grid, query and data environment based on default asset
+    default_layer = _get_default_layer(
+        dataset, default_asset.creation_options["pixel_meaning"]
+    )
+    grid = default_asset.creation_options["grid"]
+    sql = re.sub("from \w+", f"from {default_layer}", request.sql, flags=re.IGNORECASE)
+    data_environment = await _get_data_environment(grid)
+
+    input = {
+        "query": sql,
+        "environment": data_environment.dict()["layers"],
```

Comment:
In https://gfw.atlassian.net/wiki/spaces/GFWAPI/pages/1208352770/OTF+on+Lists , I think we assumed the data_environment input field to the step function will be called `data_environment`.  I think that will be clearer than just `environment`.